### PR TITLE
crypto: Add get_transaction_auth_signers

### PR DESCRIPTION
**Pull request by joyqvq**
https://github.com/MystenLabs/sui/issues/4880
- Added a new JSONRPC endpoint get_transaction_auth_signers

### CODE REVIEW

File: sdk/typescript/src/providers/json-rpc-provider.ts
Comment (Jordan-Mysten) [inline sdk/typescript/src/providers/json-rpc-provider.ts:512]: Why is this a separate RPC method vs a flag that we can call `getTransaction` with?

File: apps/explorer/src/pages/transaction-result/TransactionResult.tsx
Comment (Jordan-Mysten) [inline apps/explorer/src/pages/transaction-result/TransactionResult.tsx:106]: We need to conditionally use the API based on network support for it.

File: sdk/typescript/src/providers/json-rpc-provider.ts
Comment (joyqvq) [reply sdk/typescript/src/providers/json-rpc-provider.ts:512]: good point, we can use a flag

File: crates/sui-json-rpc/src/read_api.rs
Comment (666lcz) [inline crates/sui-json-rpc/src/read_api.rs:148]: can we avoid `unwrap` here?

File: sdk/typescript/src/providers/json-rpc-provider.ts
Comment (666lcz) [reply sdk/typescript/src/providers/json-rpc-provider.ts:512]: imo we should either
1. have the single `getTransaction` endpoint with a flag as @Jordan-Mysten suggested
2. Add a new `getAuthorityNames` endpoint but only return `Vec<AuthorityName>`. The Explorer will still use the `getTransaction` endpoint and only make a request to the new `getAuthorityNames` endpoint when the user clicks on the `Signature` tab.

I am leaning towards 2 because `getTransaction` is one of the most heavily used endpoints and fetching/including a list of authority names(especially when we have 100+ validators) in the `getTransaction` response is expensive. My hunch is that most of the Explorer users do not click on the `Signature` tab at all.

File: crates/sui-core/src/gateway_state.rs
Comment (666lcz) [inline crates/sui-core/src/gateway_state.rs:1852]: Do we need this?

File: sdk/typescript/src/providers/json-rpc-provider.ts
Comment (kchalkias) [reply sdk/typescript/src/providers/json-rpc-provider.ts:512]: I agree w Chris, let's have a different endpoint.

**Comment by joyqvq** [comment]
@666lcz im not an expert on React, but from what i see, this `TransactionResult()` is called exactly once when the url for /transaction/:id is loaded and all TxState are loaded with the RPC endpoint result from TransactionResultAPI regardless of which tab the user is viewing. while I can make a separate getAuthorityNames endpoint, it will be called the same amount of times along with the getTransaction endpoint when populating the DataType to render the frontend component. 

I also do not see there is no other place for getTransaction used in explorer, which makes me think extending the current getTransaction endpoint is OK. 

Is there a way to optionally load a react component based on which tab the user is viewing?

image

```
    if (checkIsString(id)) {
        return IS_STATIC_ENV ? (
            <TransactionResultStatic id={id} />
        ) : (
            <TransactionResultAPI id={id} />
        );
    }
```

**Comment by joyqvq** [comment]
another idea is we can make "aggregated signature" a clickable redirects that loads its auth signatures.

**Comment by 666lcz** [comment]
> Is there a way to optionally load a react component based on which tab the user is viewing? Would appreciate some help on how to refactor the code here.

cc @Jordan-Mysten could you help @joyqvq with this? I am also fine with just having an extended `getTransaction` endpoint if the frontend refactoring is too much work.

File: sdk/typescript/src/providers/json-rpc-provider.ts
Comment (kchalkias) [reply sdk/typescript/src/providers/json-rpc-provider.ts:512]: As there many opinions here I leave it to @Jordan-Mysten and @pchrysochoidis to decide. Let's get a decision by tomorrow, so we can unblock Joy.

Comment (pchrysochoidis) [reply sdk/typescript/src/providers/json-rpc-provider.ts:512]: 2. Sounds reasonable to me

Comment (joyqvq) [reply sdk/typescript/src/providers/json-rpc-provider.ts:512]: @pchrysochoidis for 2) its a bigger refactor (https://github.com/MystenLabs/sui/pull/6337#issuecomment-1328123646), unless we have more compiling use case for a more modular react components, i would prefer 1) to optionally return the auth signers key to the same getTransaction endpoint.

**Comment by joyqvq** [comment]
discussed with @Jordan-Mysten offline, we would use 2) to make a separate RPC call to load auth signers, this PR just adds the endpoint, @Jordan-Mysten will work on the pure UI PR separately. 

I realized the getCommitteeInfo endpoint is not as flexible mapping what the roaring bitmap to the committee authorities. @Jordan-Mysten I added the getTransactionAuthSigners endpoint in this PR to let rust handle the get authority by index logic. You can use this endpoint from UI to load the auth signers (with version gate)

File: crates/sui-json-rpc/src/gateway_api.rs
Comment (joyqvq) [inline crates/sui-json-rpc/src/gateway_api.rs:174]: @666lcz I believe the gateway api is being deprecated? unfortunately i had to add here to confirm the trait for read api also

**Review by 666lcz — APPROVED** [review]

File: crates/sui-open-rpc/spec/openrpc.json
Comment (kchalkias) [inline crates/sui-open-rpc/spec/openrpc.json:4408]: thanks for having this as a different response!

**Review by kchalkias — APPROVED** [review]

**Pull request merged**
